relayer/src/foreign_client.rs

```diff
@@ -875,6 +941,26 @@ impl<DstChain: ChainHandle, SrcChain: ChainHandle> ForeignClient<DstChain, SrcCh
             trusted_height
         };
 
+        // Check if the consensus state at `trusted_height` is within trusting period
+        if let ConsensusStateTrusted::NotTrusted {
+            elapsed,
+            consensus_state_timestmap,
+            network_timestamp,
+        } = self.check_consensus_state_trusting_period(&client_state, &trusted_height)?
```

Comment:
Maybe do this check only if `trusted_height != client_state.latest_height()`? This way for most of the cases we don't `query_application_status()` twice.

Reply:
ac3dbc09